internal/operator/secret.go

```diff
@@ -0,0 +1,46 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (oe operatorClient) CreateSecret(ctx context.Context, name string, content map[string]string, secretType corev1.SecretType, namespace string) (*corev1.Secret, error) {
+	secret := corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		StringData: content,
+		Type:       secretType,
+	}
+	err := oe.Client.Create(ctx, &secret, &runtimeClient.CreateOptions{})
+	if err != nil {
+		log.Error(fmt.Errorf("%w: error while creating secret: %s in namespace: %s", err, name, namespace))
+		return nil, err
+	}
+
+	log.Debugf("Secret %s created successfully in namespace %s", name, namespace)
```

Comment:
I see that we are beginning to use log a lot. Let's open an issue to replace all logrus by zap. Otherwise it will get tricky down the line. We don't need for this particular PR but right after the first ones and making everything work we gotta put that in place. I'll open that issue.